python/sqlflow_submitter/optimize/optimize.py

```diff
@@ -0,0 +1,357 @@
+# Copyright 2020 The SQLFlow Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+import numpy as np
+import pyomo.environ as pyomo_env
+import six
+from pyomo.environ import (Integers, NegativeIntegers, NegativeReals,
+                           NonNegativeIntegers, NonNegativeReals,
+                           NonPositiveIntegers, NonPositiveReals,
+                           PositiveIntegers, PositiveReals, Reals, maximize,
+                           minimize)
+
+# TODO(sneaxiy): support more aggregation functions if needed
+AGGREGATION_FUNCTIONS = ['sum']
+
+# FIXME(sneaxiy): do not know why pyomo requires that DATA_FRAME must be a global variable
+DATA_FRAME = None
+
+
+def split_to_tokens(expression):
```

Comment:
The parsed result is already tokens, so how about changing the IR `expression` from `string` to `[]string` to store tokens?

Reply:
Done.